### PR TITLE
Create an additional URL target for internal use

### PR DESCRIPTION
**Pull request by tejasd**
#no-changelog

See copybara changes in cl/326066764

### CODE REVIEW

File: GoogleDataTransport/GDTCCTLibrary/GDTCCTUploader.m [hunk 1]
```diff
@@ -157,104 +158,124 @@ - (nullable NSURL *)serverURLForTarget:(GDTCORTarget)target {
...
+          p2[27], p1[28], p2[28], p1[29], p2[29], p1[30], p2[30], p1[31], p2[31], p1[32], p2[32],
+          p1[33], p2[33], p1[34], p2[34], p1[35], p2[35], p1[36], p2[36], p1[37], p2[37], '\0'};
+      INTServerURL = [NSURL URLWithString:[NSString stringWithUTF8String:URL]];
+    });
 
 #if !NDEBUG
-  if (_testServerURL) {
-    return _testServerURL;
-  }
+    if (_testServerURL) {
```
Comment (maksymmalyhin) [inline GoogleDataTransport/GDTCCTLibrary/GDTCCTUploader.m:181]: Indentation looks wrong, could you please try to fix it with https://github.com/firebase/firebase-ios-sdk#code-formatting

File: GoogleDataTransport/GDTCCTLibrary/GDTCCTUploader.m [hunk 2]
```diff
@@ -329,55 +350,57 @@ - (void)handleURLResponse:(nullable NSURLResponse *)response
...
+      futureUploadTime = [GDTCORClock clockSnapshotInTheFuture:15 * 60 * 1000];
+    }
+    switch (target) {
+      case kGDTCORTargetCCT:
+        self->_CCTNextUploadTime = futureUploadTime;
+        break;
+
+      case kGDTCORTargetFLL:
+        // Falls through.
+      case kGDTCORTargetINT:
```
Comment (maksymmalyhin) [inline GoogleDataTransport/GDTCCTLibrary/GDTCCTUploader.m:384]: I'm not sure we should apply the next upload timeout set for another backend. is `kGDTCORTargetINT` a Firelog endpoint as well?

File: GoogleDataTransport/GDTCCTLibrary/GDTCCTUploader.m (hunk 1 above)
Comment (tejasd) [reply GoogleDataTransport/GDTCCTLibrary/GDTCCTUploader.m:181]: This ran through style.sh - but let me check.

File: GoogleDataTransport/GDTCCTLibrary/GDTCCTUploader.m (hunk 2 above)
Comment (tejasd) [reply GoogleDataTransport/GDTCCTLibrary/GDTCCTUploader.m:384]: Yes - this is for crashlytics but inside of google3.

File: GoogleDataTransport/GDTCCTLibrary/GDTCCTUploader.m (hunk 1 above)
Comment (maksymmalyhin) [reply GoogleDataTransport/GDTCCTLibrary/GDTCCTUploader.m]: Thanks, the diff looks so much better now!

File: GoogleDataTransport/GDTCORLibrary/Public/GoogleDataTransport/GDTCORTargets.h [hunk 3]
```diff
@@ -33,5 +33,8 @@ typedef NS_ENUM(NSInteger, GDTCORTarget) {
   /** The CSH target. The CSH target is a special-purpose backend. Please do not use it without
    * permission.
    */
-  kGDTCORTargetCSH = 1002
+  kGDTCORTargetCSH = 1002,
+
+  /** The INT target. */
+  kGDTCORTargetINT = 1003,
```
[the remark this answers is not in the document]
Comment (tejasd) [reply GoogleDataTransport/GDTCORLibrary/Public/GoogleDataTransport/GDTCORTargets.h]: Let me know if the cc_scrub looks acceptable!

File: GoogleDataTransport/GDTCCTLibrary/GDTCCTUploader.m [hunk 4]
```diff
@@ -175,14 +183,19 @@ - (nullable NSURL *)serverURLForTarget:(GDTCORTarget)target {
     case kGDTCORTargetCSH:
       return CSHServerURL;
 
+      // copybara:insert_begin(Reserve private endpoint)
+      //    case kGDTCORTargetINT:
```
Comment (mikehaney24) [inline GoogleDataTransport/GDTCCTLibrary/GDTCCTUploader.m:187]: in the sample import, this case is indented 4 too many, so probably `// case kGDTCORTargetInt:`

Comment (tejasd) [reply GoogleDataTransport/GDTCCTLibrary/GDTCCTUploader.m]: Fixed.

**Comment by tejasd** [comment]
I'll submit this with cc_scrub for now - feel free to delete it and switch it to just replacing `kINTServerURL` if it becomes a hassle during refactoring (shouldn't require a copybara change if done so).

**Pull request merged**